ubi_manifest/worker/tasks/depsolve.py

```diff
@@ -157,22 +175,63 @@ def depsolve_task(ubi_repo_ids: Iterable[str], content_config_url: str) -> None:
         debuginfo_out = _run_depsolver(
             list(debug_dep_map.values()),
             repos_map,
-            in_source_rpm_repos,
             modulemd_rpm_deps,
             modular_rpm_filenames,
             flags,
         )
 
-    # merge 'out' and 'debuginfo_out' dicts without overwriting any entry
-    _merge_output_dictionary(out, debuginfo_out)
+        _get_srpm_filenames(client, debuginfo_out, srpm_filenames)
+
+        # save debug manifests to redis and delete data
+        _save(debuginfo_out)
+        del debuginfo_out
+
+        # run depsolver for source repos
+        _LOG.info(
+            "Running depsolver for SRPM repos: %s",
+            [item[0] for item in srpm_blacklist_map],
+        )
+        srpm_out = _run_srpm_depsolver(
+            list(srpm_blacklist_map.values()),
+            srpm_filenames,
+            in_source_rpm_repos,
+            repos_map,
+        )
+        _save(srpm_out)
+
+        _ensure_all_manifests(list(repos_map.values()))
+
+
+def _get_srpm_filenames(
+    client: Client, data: dict[str, list[UbiUnit]], srpm_filenames: dict[str, list[str]]
+) -> None:
+    """
+    Gets SRPM filenames from all passed binary/debug RPMs. Resulting srpm_filenames
+    dict is modified in place, with respective rhel-source-repos as keys, so it is
+    possible to search for the SRPMs directly in the repos where they are expected.
+    """
+    for units in data.values():
+        # `data` are in the format of `"UBI binary/debug repo": [UbiUnits]`, however, we
+        # need to get their respective rhel source repos, where we can search for the SRPMs.
+        # Because client-tools repos have usually two rhel repos as population sources,
```

Comment:
we shouldn't probably refer to `client-tools` as it's only internal naming and in general case any output (UBI) repos can have more than one population source.

...also because we don't use that only for UBI anymore, we shouldn't even refer to it (much :) ) - so maybe in future we should rename it: UBI  repo -> output repo; RHEL repo -> input repo. But that's outside of scope of this change.